src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -1056,7 +1056,17 @@ private void setSuperInterfaceDerivedBoundingTypeOnWildcard(
 					"Ignoring type parameters for problem binding: " + binding);
 		}
 
-		String simpleName = readableName.substring(Math.max(0, readableName.lastIndexOf('.') + 1));
+		// cut package name, assuming package names start with a lowercase letter
+		String n = readableName;
+		while (Character.isLowerCase(n.charAt(0))) {
+			int idx = n.indexOf('.');
+			if (idx < 0) {
+				break;
+			}
+			n = n.substring(idx + 1);
+		}
```

Comment:
```suggestion
		String fullClassName = stripPackageName(readableName)
```
Let's not use `simpleName` now because that refers to just the name of the class and does not include its parent types. Maybe something like `fullClassName` is clearer. I also suggest abstracting this loop to a separate function just for the sake of readability.

```java
private static String stripPackageName(String fullyQualifiedName) {
	if (Character.isLowerCase(fullyQualifiedName.charAt(0))) {
		int indexToInnerPackage = fullyQualifiedName.indexOf('.');
		return stripPackageName(fullyQualifiedName.substring(indexToInnerPackage + 1));
	}
	return fullyQualifiedName;
}
```
I thought a recursive call might be clearer to read and that is why I refactored like above. Note that I removed the condition `idx < 0` because you are writing this function with an assumption "assuming package names start with a lowercase letter" so if the first character is indeed a lowercase, you will never reach this condition because you are traversing a package right now and types are yet to follow _after the delimiter_.

Reply:
I agree that this should be extracted into a separate function. Whether or not it should be recursive is a matter of taste, I think @xzel23 can use their preference.